source/dub/internal/vibecompat/core/file.d

```diff
@@ -156,8 +156,56 @@ private bool sameFile(NativePath a, NativePath b)
 	}
 }
 
+private bool isWritable(NativePath name)
+{
+	version (Windows)
+	{
+		import core.sys.windows.windows;
+
+		return (name.toNativeString.getAttributes & FILE_ATTRIBUTE_READONLY) == 0;
+	}
+	else version (Posix)
+	{
+		import core.sys.posix.sys.stat;
+
+		return (name.toNativeString.getAttributes & S_IWUSR) != 0;
+	}
+	else
+		static assert("Needs implementation.");
+}
+
+private void makeWritable(NativePath name)
+{
+	makeWritable(name.toNativeString);
+}
+
+private void makeWritable(string name)
+{
+	version (Windows)
+	{
+		import core.sys.windows.windows;
+
+		name.setAttributes(name.getAttributes & ~FILE_ATTRIBUTE_READONLY);
+	}
+	else version (Posix)
+	{
+		import core.sys.posix.sys.stat;
+
+		name.setAttributes(name.getAttributes | S_IWUSR);
+	}
+	else
+		static assert("Needs implementation.");
+
+	import std.exception;
+	import std.stdio;
+	assertNotThrown!ErrnoException(File(name, "w"));
```

Comment:
```suggestion
	version(assert) assertNotThrown!ErrnoException(File(name, "a"));
```

didn't this clear the output file in testing? imo this line is dangerous and might also be removed completely.

Reply:
Well spotted. Yes it probably cleared the output file, but since that was already 0 bytes in length, there was no change. I'll remove the line completely.